examples/ios/swift/ListSwiftUI/Views/ContentView.swift

```diff
@@ -1,126 +1,148 @@
-            }
-            Spacer()
-        }
+    public static func ==(lhs: Dog, rhs: Dog) -> Bool {
+        return lhs.isSameObject(as: rhs)
     }
 
-    private func isExpanded(_ section: Recipe) -> Bool {
-        self.state.sectionState[section] ?? false
-    }
+    let handlers = LinkingObjects(fromType: Person.self, property: "dogs")
 }
 
-final class ContentViewState: ObservableObject {
-    /// This dict will allow us to store state on the expansion and contraction of rows.
-    @Published var sectionState: [Recipe: Bool] = [:]
+// MARK: Person Model
+class Person: Object, ObjectKeyIdentifiable {
+    private static let peopleNames = [
+        "Aoife", "Caoimhe", "Saoirse", "Ciara", "Niamh",
+        "Conor", "Seán", "Oisín", "Patrick", "Cian",
+        "Isabella", "Mateo", "Emilia", "Savannah", "Isla",
+        "Elena", "Maya", "Santiago", "Gabriella", "Leonardo"
+    ]
+    /// The unique id of this dog
+    @objc dynamic var id = ObjectId.generate()
+    /// The name of the person
+    @objc dynamic var name = peopleNames.randomElement()!
+    /// The dogs this person has
+    var dogs = RealmSwift.List<Dog>()
 }
 
-struct ContentView: View {
-    @State private var searchTerm: String = ""
-    @State private var showRecipeFormView = false
-    @ObservedObject var state = ContentViewState()
-    @ObservedObject var recipes: RealmSwift.List<Recipe>
+struct DogList: View {
+    @RealmState var dogs: RealmSwift.List<Dog>
 
     var body: some View {
-        NavigationView {
-            List {
-                Section(header:
-                    HStack {
-                        SearchBar(text: self.$searchTerm)
-                            .frame(width: 300, alignment: .leading)
-                            .padding(5)
-                        NavigationLink(destination: RecipeFormView(recipes: self.recipes, showRecipeFormView: self.$showRecipeFormView),
-                                       isActive: self.$showRecipeFormView,
-                                       label: {
-                                        Button("add recipe") {
-                                            self.showRecipeFormView = true
-                                        }
-                        })
-                }) {
-                    ForEach(filteredCollection().freeze()) { recipe in
-                        RecipeRow(recipe).environmentObject(self.state)
-                    }
-                    .onDelete(perform: delete)
-                    .onMove(perform: move)
-                }
-            }.listStyle(GroupedListStyle())
-                .navigationBarTitle("recipes", displayMode: .large)
-                .navigationBarBackButtonHidden(true)
-                .navigationBarHidden(false)
-                .navigationBarItems(trailing: EditButton())
+        List {
+            // Using the `$` will bind the Dog List to the view
+            ForEach(dogs) { dog in
+                TextField("dog name", text: bind(dog, \.name))
```

Comment:
will this `bind` API be hard to discover? Do we also want to extend `View` to do the also?

Reply:
My other thought was to put `bind` on `ThreadConfined` types, but I'm not sure if there is value in that outside of SwiftUI (it's API bloat otherwise, but maybe that's fine?). I'm open to ideas anyways.

Reply:
I'm just thinking we should extend our own domain and not another party's. But it's tricky because its SwiftUI specific. I guess I'm fine with extending `View` as its a very niche case

Reply:
I actually don't think this is terribly niche, which is why I'm trying to be careful– binding to a SwiftUI view will likely be fairly common (TextFields, sliders, etc.).

Thoughts @tgoyne ?